### PR TITLE
Apply black to the codebase

### PR DESCRIPTION
**Pull request by ericof**
* Implement lint checking with flake8, isort, black, readme_renderer and check-manifest
* Apply black to the codebase

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    # E203: whitespace before ':' (black fails to be PEP8 compliant)
+    E203
+    # E731: do not assign a lambda expression, use a def
+    E731
+    # W503: line break before binary operator (flake8 is not PEP8 compliant)
+    W503
+exclude =
+    docs
+show-source = True
+max-line-length = 88
```
Comment (stevepiercy) [inline .flake8]: 79.

Add carriage return at EOF.

File: .isort.cfg [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+known_first_party=deform
+import_heading_stdlib=Standard Library
+import_heading_pyramid=Pyramid
+import_heading_firstparty=Deform
+lines_after_imports = 2
+not_skip=__init__.py
+include_trailing_comma=True
+force_grid_wrap=0
+combine_as_imports=True
+line_length=88
```
Comment (stevepiercy) [inline .isort.cfg]: 79.

Add carriage return at EOF.

File: .travis.yml [hunk 3]
```diff
@@ -31,6 +31,8 @@ matrix:
       env: TOXENV=py2-cover,py3-cover,coverage
     - python: "3.6"
       env: TOXENV=docs
+    - python: "3.6"
+      env: TOXENV=lint
```
Comment (stevepiercy) [inline .travis.yml]: Add python 3.7. [See Pyramid's](https://github.com/Pylons/pyramid/blob/master/.travis.yml#L25-L34):

```
        - python: 3.7
          env: TOXENV=py37
          dist: xenial
          sudo: true
        - python: 3.8-dev
          env: TOXENV=py38
          dist: xenial
          sudo: true
    allow_failures:
        - env: TOXENV=py38
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[tool.black]
+line-length = 88
```
Comment (stevepiercy) [inline pyproject.toml]: 79

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+include = '\.pyi?$'
+exclude = '''
+/(
+    \.git
+)/
+'''
```
Comment (stevepiercy) [inline pyproject.toml]: Add carriage return at EOF.

File: setup.py [hunk 6]
```diff
@@ -12,59 +12,62 @@
...
+    "check-manifest",
+    "flake8",
+    "flake8-bugbear",
+    "flake8-builtins",
+    "flake8-isort",
+    "isort",
+    "readme_renderer",
 ]
 
+testing_extras = ["nose", "coverage", "beautifulsoup4", "flaky"]
```
Comment (stevepiercy) [inline setup.py]: one line per item

File: setup.py [hunk 7]
```diff
@@ -12,59 +12,62 @@
...
+
+# Needed to run deformdemo tests
+functional_testing_extra = ["selenium<3", "pyramid", "pygments", "waitress", "lingua"]
```
Comment (stevepiercy) [inline setup.py]: one line per item

File: .travis.yml (hunk 3 above)
Comment (ericof) [reply .travis.yml]: This is only for the lint

**Pull request merged**